wandb/sdk/wandb_run.py

```diff
@@ -274,13 +273,16 @@ def _process_network_status(result: Result) -> None:
         )
 
     def check_stop_status(self) -> None:
+        import signal
+
         def _process_stop_status(result: Result) -> None:
             stop_status = result.response.stop_status_response
             if stop_status.run_should_stop:
                 # TODO(frz): This check is required
                 # until WB-3606 is resolved on server side.
                 if not wandb.agents.pyagent.is_running():
-                    thread.interrupt_main()
+                    # thread.interrupt_main()
+                    os.kill(os.getpid(), signal.SIGINT)
```

Comment:
Do we know why `thread.interrupt_main()` wasn't working? It was a little hard to figure out, but is it because Python won't raise the KeyboardInterrupt if the main thread is busy in some low-level code until it's done?

But that doesn't explain why `os.kill` works. It [says](https://docs.python.org/3/library/os.html#os.kill) it sends a SIGINT to the process, which would then be handled by the main thread, so I feel like the effect ought to be the same.

Reply:
yeah exactly, i was just experimenting with it... will need to dig to figure out, why thread.interrupt_main() didn't work :(